app/uk/gov/hmrc/partnershipidentificationfrontend/api/controllers/JourneyController.scala

```diff
@@ -51,15 +51,8 @@ class JourneyController @Inject()(controllerComponents: ControllerComponents,
 
   def createLimitedLiabilityPartnershipJourney: Action[JourneyConfig] = createJourney(LimitedLiabilityPartnership)
 
-  private def createJourney(partnershipType: PartnershipType): Action[JourneyConfig] =
-    Action.async(parse.json[JourneyConfig] { json =>
-      for {
-        continueUrl <- (json \ continueUrlKey).validate[String]
-        optServiceName <- (json \ optServiceNameKey).validateOpt[String]
-        deskProServiceId <- (json \ deskProServiceIdKey).validate[String]
-        signOutUrl <- (json \ signOutUrlKey).validate[String]
-      } yield JourneyConfig(continueUrl, PageConfig(optServiceName, deskProServiceId, signOutUrl), partnershipType)
-    }) {
+  private def createJourney(partnershipType: PartnershipType): Action[JourneyConfig] = {
+    Action.async(fromJsonToAJourneyConfig(partnershipType)) {
```

Comment:
Would it be better just to use

Action.async(parse.json[JourneyConfig](reader = journeyConfigJsonReader.toJourneyConfig(partnershipType)))

rather than call the private method "fromJsonToAJourneyConfig"?

Reply:
not sure Richard I did it as first option and after I changed because the second one it reads better ... really dont know .... you decide

Reply:
hey Richard I named the parameter ... now it is :
Action.async(bodyParser = fromJsonToAJourneyConfig(partnershipType)) .... and I prefer it:
1) clear
2) readable

Reply:
fromJsonToAJourneyConfig(partnershipType) is removed and Action.async(parse.json[JourneyConfig] is used so its resolved